api/common/status_types.go

```diff
@@ -0,0 +1,25 @@
+package common
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+const (
+	// StatusPhaseReady indicates that the resource is ready. All conditions are met and are in status "True".
+	StatusPhaseReady = "Ready"
+	// StatusPhaseProgressing indicates that the resource is not ready and being created or updated. At least one condition is not met and is in status "False".
+	StatusPhaseProgressing = "Progressing"
+	// StatusPhaseTerminating indicates that the resource is not ready and in deletion. At least one condition is not met and is in status "False".
+	StatusPhaseTerminating = "Terminating"
+)
+
+// Status represents the status of an openMCP resource.
+type Status struct {
+	// ObservedGeneration is the generation of this resource that was last reconciled by the controller.
+	ObservedGeneration int64 `json:"observedGeneration"`
+
+	// Phase is the current phase of the resource.
+	Phase string `json:"phase"`
+
+	// Conditions contains the conditions.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
```

Comment:
How does the `Status` struct here relates to the `CommonStatus` struct in the `shared_types.go` file? Can we already get rid of ...

```golang
// Conditions contains the conditions.
	// +optional
	Conditions ConditionList `json:"conditions,omitempty"`
```

... with this PR?!

Reply:
Ah, I just added the reason for this in the PR description, but it was not there originally.
Basically, the controller-utils lib has some functions which depend on the current structure and some controllers in this repo use these functions. To avoid having 'broken' commits, I would prefer to do this in multiple steps - add new stuff here, adapt controller-utils to new stuff, replace old stuff with new stuff here.